crates/rust-client/src/accounts.rs

```diff
@@ -17,11 +23,29 @@ use winter_maybe_async::{maybe_async, maybe_await};
 use super::{rpc::NodeRpcClient, Client};
 use crate::{store::Store, ClientError};
 
+/// Defines templates for creating different types of Miden accounts.
 pub enum AccountTemplate {
+    /// The `BasicWallet` variant represents a regular wallet account.
+    ///
+    /// # Fields
+    ///
+    /// - `mutable_code` - A boolean indicating whether the account's code can be modified after
+    ///   creation.
+    /// - `storage_type` - Specifies the type of storage used by the account. This is defined by
+    ///   the `AccountStorageType` enum.
     BasicWallet {
         mutable_code: bool,
         storage_type: AccountStorageType,
     },
```

Comment:
Makes sense to move the Fields docs inside the struct? Like:
```suggestion
    /// The `BasicWallet` variant represents a regular wallet account.
    BasicWallet {
        /// - `mutable_code` - A boolean indicating whether the account's code can be modified after
        ///   creation.
        mutable_code: bool,
        /// - `storage_type` - Specifies the type of storage used by the account. This is defined by
        ///   the `AccountStorageType` enum.
        storage_type: AccountStorageType,
    },
```

Reply:
Good point as well, changed